.circleci/config.yml

```diff
@@ -1,18 +1,113 @@
 version: 2.1
 
+# add your orb below, to be used in integration tests (note: a @dev:alpha
+# release must exist.);
+orbs:
+  aws-code-deploy: circleci/aws-code-deploy@dev:alpha
+  aws-cli: circleci/aws-cli@0.1
+  orb-tools: circleci/orb-tools@9.0
+
+# Pipeline parameters
+parameters:
+  # These pipeline parameters are required by the "trigger-integration-tests-workflow"
+  # job, by default.
+  run-integration-tests:
+    type: boolean
+    default: false
+  dev-orb-version:
+    type: string
+    default: "dev:alpha"
+
 jobs:
-  getting-started:
+  # This job is an example of an integration testing job.
+  # This job should execute a command from your orb and verify
+  # the output is as expected, otherwise the job should fail.
+  #
+  # Rename this job and add more as needed.
+  #
+  integration-test-1:
     docker:
-      - image: circleci/node
+      - image: cimg/python:3.8
     steps:
+      - checkout
+      - aws-cli/setup
       - run:
-          name: Getting Started
-          command: |
-            echo "Your orb is now building on CircleCI"
-            echo "This is a temporary configuration file. Continue with the orb-init.sh script and follow the README.md to publish your orb."
+          name: Start Instance
+          command: aws ec2 start-instances --instance-ids "i-07639163720fe5338"
+      - aws-code-deploy/push-bundle:
+          application-name: CCI_CodeDeploy_Orb
+          bundle-source: ./sample_app
+          bundle-bucket: orb-testing
+          bundle-key: code_deploy_sample
+      - aws-code-deploy/deploy-bundle:
+          application-name: CCI_CodeDeploy_Orb
+          deployment-group: CCI_Orb_Test
+          bundle-bucket: orb-testing
+          bundle-key: code_deploy_sample
+      - run:
+          name: Shutdown Instance
+          command: aws ec2 stop-instances --instance-ids "i-07639163720fe5338"
+
 workflows:
-  setup:
+  # This `lint-pack_validate_publish-dev` workflow will run on any commit.
+  lint_pack-validate_publish-dev:
+    unless: << pipeline.parameters.run-integration-tests >>
+    jobs:
+      - orb-tools/lint
+      # pack your orb YAML files to a single orb.yml
+      # validate the orb.yml file to ensure it is well-formed
+      - orb-tools/pack:
+          requires:
+            - orb-tools/lint
+
+      # release dev version of orb, for testing & possible publishing.
+      # orb will be published as dev:alpha and dev:${CIRCLE_SHA1:0:7}.
+      # requires a CircleCI API token to be stored as CIRCLE_TOKEN (default)
+      # https://circleci.com/docs/2.0/managing-api-tokens
+      # store CIRCLE_TOKEN as a project env var or Contexts resource
+      # if using Contexts, add your context below
+      - orb-tools/publish-dev:
+          orb-name: circleci/aws-code-deploy
```

Comment:
I think it'd be good if we can add the `context: orb-publishing` here so that we wouldn't need to also define `CIRCLE_TOKEN` in the project environment variables